ssz/sedes/tuple.py

```diff
@@ -0,0 +1,108 @@
+from typing import (
+    Generator,
+    Sequence,
+    Tuple,
+    TypeVar,
+)
+
+from eth_utils import (
+    to_tuple,
+)
+
+from ssz.utils import (
+    merkleize,
+    pack,
+)
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes.base import (
+    BaseSedes,
+    BasicSedes,
+    CompositeSedes,
+)
+
+TSerializableElement = TypeVar("TSerializable")
+TDeserializedElement = TypeVar("TDeserialized")
+
+
+class Tuple(CompositeSedes[Sequence[TSerializableElement], Tuple[TDeserializedElement, ...]]):
+
+    def __init__(self,
+                 number_of_elements: int,
+                 element_sedes: BaseSedes[TSerializableElement, TDeserializedElement]) -> None:
+
+        self.number_of_elements = number_of_elements
+        self.element_sedes = element_sedes
+
+    #
+    # Size
+    #
+    @property
+    def is_variable_length(self):
+        return self.number_of_elements > 0 and self.element_sedes.is_variable_length
+
+    def get_fixed_length(self):
+        if self.is_variable_length:
+            raise ValueError("Tuple does not have a fixed length")
```

Comment:
Based on this message I think we've got a naming issue.  The name *length* to me implies *how many elements are in the thing*, but this seems to be using it as *what size is the encoded thing*.  IIUC correctly, a `Tuple` represents a fixed size array of homogenous things, and thus, should always be a fixed length (number of elements).  That suggests to me we should be using the name `size` for this, something like `is_dynamic_sized`

Reply:
Agreed, using size for "number of bytes in a string" and length for "number of elements in (the first level of) a composite object" would be much cleaner. Will update the PR. Probably a good idea to use that terminology in the spec.